### PR TITLE
feat: migrate admin/store to typescript

### LINKED ISSUE #754: Convert to typescript - admin/store

### PR DESCRIPTION
**Pull request by zakariaelas**
fixes #754

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/store/update-store.ts [hunk 1]
```diff
@@ -32,20 +34,31 @@ import { MedusaError, Validator } from "medusa-core-utils"
...
-  const store = await storeService.update(value)
+  const storeService: StoreService = req.scope.resolve("storeService")
+  const store = await storeService.update(validatedBody)
   res.status(200).json({ store })
 }
+
+export class AdminPostStoreReq {
+  @IsOptional()
+  @IsString()
+  name: string
```
Comment (sebastiannicolajsen) [inline packages/medusa/src/api/routes/admin/store/update-store.ts:47]: Missing `?` after name (`name?: string`)

File: packages/medusa/src/api/routes/admin/store/update-store.ts [hunk 2]
```diff
@@ -32,20 +34,31 @@ import { MedusaError, Validator } from "medusa-core-utils"
...
+
+  @IsString()
+  @IsOptional()
+  swap_link_template: string
+
+  @IsString()
+  @IsOptional()
+  payment_link_template: string
```
Comment (sebastiannicolajsen) [inline packages/medusa/src/api/routes/admin/store/update-store.ts:55]: Same as the above.

File: packages/medusa/src/api/routes/admin/store/update-store.ts [hunk 3]
```diff
@@ -32,20 +34,31 @@ import { MedusaError, Validator } from "medusa-core-utils"
...
+
+  @IsString()
+  @IsOptional()
+  default_currency_code: string
```
Comment (sebastiannicolajsen) [inline packages/medusa/src/api/routes/admin/store/update-store.ts:59]: Same as the above.

File: packages/medusa/src/api/routes/admin/store/update-store.ts [hunk 4]
```diff
@@ -32,20 +34,31 @@ import { MedusaError, Validator } from "medusa-core-utils"
...
+
+  @IsString({ each: true })
```
[the remark this answers is not in the document]
Comment (sebastiannicolajsen) [reply packages/medusa/src/api/routes/admin/store/update-store.ts:61]: In addition to the `@IsString({ each: true})`

**Review by sebastiannicolajsen — COMMENTED** [review]
LGTM

**Review by srindom — APPROVED** [review]

**Pull request merged**